### PR TITLE
Warn on 'await await'

### LINKED ISSUE #776: 7$: Warn on 'await await'
In an async function, `await await p` is no different than `await p`. Mark the first `await` as redundant.

### PR DESCRIPTION
**Pull request by clegoz**
Resolves #776

### CODE REVIEW

**Comment by clegoz** [comment]
I have read and hereby agree to quick-lint-js' Contributor License Agreement (CLA-v1.md).

File: src/quick-lint-js/diagnostic-types.h [hunk 1]
```diff
@@ -1980,6 +1980,11 @@
       MESSAGE(QLJS_TRANSLATABLE("continue can only be used inside of a loop"), \
               continue_statement))                                             \
                                                                                \
+  QLJS_DIAG_TYPE(                                                              \
+      diag_redundant_await, "E0260", diagnostic_severity::warning,             \
```
Comment (strager) [inline src/quick-lint-js/diagnostic-types.h:1984]: Nit: Sort alphabetically by struct name (`diag_redundant_await`).

File: test/test-parse-expression.cpp [hunk 2]
```diff
@@ -886,6 +886,28 @@ TEST_F(test_parse_expression, await_unary_operator_inside_async_functions) {
     EXPECT_EQ(summarize(ast), "await(paren(var x))");
     EXPECT_THAT(p.errors(), IsEmpty());
   }
+
+  {
```
Comment (strager) [inline test/test-parse-expression.cpp:890]: Nit: I'd put this in a separate `TEST` so it's better documented.

File: test/test-parse-expression.cpp [hunk 3]
```diff
@@ -1126,7 +1149,9 @@ TEST_F(test_parse_expression,
       auto guard = p.parser().enter_function(function_attributes::async);
       expression* ast = p.parse_expression();
       EXPECT_EQ(summarize(ast), test.expected_async_function);
-      EXPECT_THAT(p.errors(), IsEmpty());
+
+      EXPECT_THAT(p.errors(), ::testing::Not(::testing::Contains(DIAG_TYPE(
+                                  diag_await_operator_outside_async))));
```
Comment (strager) [inline test/test-parse-expression.cpp:1154]: Must fix: Negative assertions are bad and we should avoid them if possible.

If the diagnostics change for only one test case, we should special-case that test case (similar to what we do on line 1126).

File: test/test-parse-expression.cpp [hunk 4]
```diff
@@ -1136,7 +1161,8 @@ TEST_F(test_parse_expression,
       EXPECT_EQ(summarize(ast), test.expected_async_function
                                     ? test.expected_async_function
                                     : test.expected_normal_function);
-      EXPECT_THAT(p.errors(), IsEmpty());
+      EXPECT_THAT(p.errors(), ::testing::Not(::testing::Contains(DIAG_TYPE(
+                                  diag_await_operator_outside_async))));
```
Comment (strager) [inline test/test-parse-expression.cpp:1165]: Must fix: Same comment as on line 1153-4.

File: test/test-parse.cpp [hunk 5]
```diff
@@ -750,7 +750,8 @@ TEST(test_no_overflow, parser_depth_limit_not_exceeded) {
     parser p(&code, &v);
     bool ok = p.parse_and_visit_module_catching_fatal_parse_errors(v);
     EXPECT_TRUE(ok);
-    EXPECT_THAT(v.errors, IsEmpty());
+    EXPECT_THAT(v.errors, ::testing::Not(::testing::Contains(
+                              DIAG_TYPE(diag_depth_limit_exceeded))));
```
Comment (strager) [inline test/test-parse.cpp:754]: Note: Generally, negative assertions are bad. In this case, I think it's fine. You can leave this code as-is.

**Review by strager — CHANGES_REQUESTED** [review]

**Comment by strager** [comment]
I fixed the "Homebrew package / macOS 10.15" CI failure on master. Please rebase/merge.

**Review by strager — APPROVED** [review]

**Pull request closed without merging**